lib/Pipes/TaggedFunctionKind.cpp

```diff
@@ -50,6 +52,107 @@ using TaggedFK = TaggedFunctionKind;
 void TaggedFK::getInvalidations(const Context &Ctx,
                                 TargetsList &ToRemove,
                                 const GlobalTupleTreeDiff &Diff) const {
+  return;
+}
+
+using IsolatedFK = IsolatedFunctionKind;
+void IsolatedFK::getInvalidations(const Context &Ctx,
+                                  TargetsList &ToRemove,
+                                  const GlobalTupleTreeDiff &Diff) const {
+  using PM = PathMatcher;
+  using Binary = model::Binary;
+
+  const auto &CurrentModel = getModelFromContext(Ctx);
+  if (not Ctx.containsReadOnlyContainer(BinaryCrossRelationsRole))
+    return;
+
+  static constexpr auto BCRR = BinaryCrossRelationsRole;
+  const auto &Container = Ctx.getReadOnlyContainer<FileContainer>(BCRR);
+
+  auto MaybeContainerPath = Container.path();
+  revng_assert(MaybeContainerPath.has_value());
+  auto MaybeBuffer = llvm::MemoryBuffer::getFile(*MaybeContainerPath);
+  revng_assert(MaybeBuffer);
+  yaml::Input YAMLInput((*MaybeBuffer)->getBuffer());
+
+  yield::CrossRelations Relations;
+  YAMLInput >> Relations;
+
+  const auto *ModelDiff = Diff.getAs<Binary>();
+  if (not ModelDiff)
+    return;
+
+  const Binary &Model = *CurrentModel.get();
+  auto EntryMatcher = PM::create<Binary>("/Functions/*/Entry").value();
```

Comment:
As we said multiple times, the key *cannot* change. You should never see this.
If someone changes the key, you will not see a change on `/Functions/0x1000/Entry` you will see the whole function (`/Functions/0x1000`) being removed and added again (`/Functions/0x2000`).
Did you to make such a change and see what `revng model diff` does?

Reply:
I realize it, but the regex `"/Functions/*/Entry"` used to work correctly to match an addition and a removal until not a long ago. That said, since not correct, I had changed it. The approach used now should be correct.